provisioning/roles/php/tasks/debian-repo.yml

```diff
@@ -1,24 +1,38 @@
 - name: debian-repo | error when trying to use something less than PHP 5.6 on jessie
   fail: msg="Jessie only supports version 5.6."
-  when: "ansible_lsb.major_release|int >= 8 and {{ php_version_installed }} < 5.6"
+  when: "ansible_lsb.major_release|int >= 8 and {{ php_version_installed | version_compare('5.6', '<') }}"
 
-- name: debian-repo | Install DotDeb key
+- name: debian-repo | error when trying to use something less than PHP 7 on Stretch
+  fail: msg="Stretch only supports version 7"
+  when: "ansible_lsb.major_release|int >= 9 and {{ php_version_installed | version_compare('7.0', '<') }}"
+
+- name: debian-repo | Install DotDeb key on jessie
   apt_key: url=https://www.dotdeb.org/dotdeb.gpg
   become: yes
-  when: "{{ php_version_installed }} > 5.4"
+  when: "ansible_lsb.major_release|int <= 8 and {{ php_version_installed | version_compare('5.4', '>') }}"
 
 - name: debian-repo | Make sure APT supports HTTPS sources
   apt: pkg=apt-transport-https state=present
   become: yes
 
-- name: debian-repo | Install deb.sury.org repository key
+- name: debian-repo | Install packages.sury.org repository key on jessie/stretch
   apt_key: id=AC0E47584A7A714D url=https://packages.sury.org/php/apt.gpg state=present
-  when: "{{ php_version_installed| version_compare('7.0', '>=') }}"
+  when: "ansible_lsb.major_release|int >= 7 and {{ php_version_installed| version_compare('7.0', '>=') }}"
   become: yes
 
-- name: debian-repo | Add deb.sury.org repository for PHP 7.x on Jessie
+- name: debian-repo | Add packages.sury.org repository for PHP 7.x on Debian
   apt_repository: repo='{{ item }}' update_cache=yes
   with_items:
-    - deb https://packages.sury.org/php jessie main
+    - deb https://packages.sury.org/php {{ ansible_distribution_release }} main
+  become: yes
+  when: "ansible_lsb.major_release|int >= 7 and {{ php_version_installed| version_compare('7.0', '>=') }}"
+
+- name: debian-repo | Mark packages.sury.org repository for PHP 7.x on Debian as lower priority
+  copy:
+    dest: /etc/apt/preferences.d/packages_sury_org
+    content: |
+      Package: *
+      Pin: origin "packages.sury.org"
+      Pin-Priority: 100
   become: yes
-  when: "{{ php_version_installed| version_compare('7.0', '>=') }}"
+  when: "ansible_lsb.major_release|int >= 7 and {{ php_version_installed| version_compare('7.0', '>=') }}"
```

Comment:
It would be better to extract all these sury tasks to an external task and include it conditionnally so that you don't need to repeat the condition 3 times.